src/solc_0.5/Land/erc721/LandBaseTokenV3.sol

```diff
@@ -154,12 +244,72 @@ contract LandBaseTokenV3 is ERC721BaseTokenV2 {
         _checkBatchReceiverAcceptQuad(msg.sender, address(0), to, size, x, y, data);
     }
 
-    function _idInPath(uint256 i, uint256 size, uint256 x, uint256 y) internal pure returns(uint256) {
+    function checkAndClearOwner(
+        uint256 size,
+        uint256 x,
+        uint256 y,
+        Land[] memory mintedLand,
+        uint256 index,
+        uint256 quadCompareSize
+    ) internal {
+        (, uint256 layer, , ) = _getQuadInfo(quadCompareSize, 0);
+        uint256 toX = x + size;
+        uint256 toY = y + size;
+
+        if (size >= quadCompareSize) {
+            for (uint256 xi = x; xi < toX; xi += quadCompareSize) {
+                for (uint256 yi = y; yi < toY; yi += quadCompareSize) {
+                    bool isQuadChecked = isQuadCheckedForOwner(mintedLand, xi, yi, quadCompareSize, index);
+                    if (!isQuadChecked) {
+                        uint256 id = layer + xi + yi * GRID_SIZE;
```

Comment:
can we replace all the id calculations by a function ?
`uint256 id = layer + xi + yi * GRID_SIZE;` =>

```
uint256 id = _getQuadId(size, x, y);
```